src/lib/utils/npm-user-agent-parser.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Parses a user agent string for NPM metadata.
+ * @param {string} userAgent 
+ * @return {{npmVersion: string, nodeVersion: string, platform: string, architecture: string}}
+ */
+const npmUserAgentParser = (userAgent) => {
```

Comment:
Well, as is your idea, you can decide. I've been moving out some reusable pieces of code to separete repositories and dependencies eg: https://github.com/verdaccio/file-locking . I'm doing it because future plugins will need it. In your hands the decision.